src/csharp/Microsoft.Spark/Utils/CommandSerDe.cs

```diff
@@ -299,5 +308,45 @@ private static Delegate CreateUdfWrapperDelegate<T>(Type type, object[] paramete
                 udfWrapper,
                 UdfWrapperMethodName);
         }
+
+        private static Assembly SerDeResolveEventHandler(object sender, ResolveEventArgs args) =>
+            s_assemblyCache.GetOrAdd(args.Name, asm => LoadAssembly(args.Name));
+
+        /// <summary>
+        /// Returns the loaded assembly by probing the following locations in order:
+        /// 1) The working directory
+        /// 2) The directory of the application
+        /// If the assembly is not found in the above locations, the exception from
+        /// Assembly.LoadFrom() will be propagated.
+        /// </summary>
+        /// <param name="assemblyName">The FullName of the assembly to load</param>
+        /// <returns>The loaded assembly</returns>
+        /// <exception cref = "System.IO.FileNotFoundException" > Thrown when the FullName of
+        /// the loaded assembly does not match the FullName of the assembly to load.</exception>
+
+        private static Assembly LoadAssembly(string assemblyName)
+        {
+            var sep = Path.DirectorySeparatorChar;
+            var asmSimpleName = assemblyName.Split(',').FirstOrDefault();
+            Assembly asm;
+            try
+            {
+                asm = Assembly.LoadFrom(
```

Comment:
Have you checked out https://docs.microsoft.com/en-us/dotnet/framework/deployment/best-practices-for-assembly-loading? I don't think `Assembly.LoadFrom` is recommended anymore. But instead, using AssemblyLoadContext.

/cc @stephentoub

Reply:
See also the way MSBuild loads assemblies for an example:

https://github.com/microsoft/msbuild/blob/fc10ea8ce260b764bb9fa5033b327af9fefcaabe/src/Shared/TaskEngineAssemblyResolver.cs#L127

Reply:
Thanks for the suggestion and finding the article.  For now we will create issue #138 and revisit the assembly loading in another PR.

Reply:
@eerhardt I investigated this a bit more.  The AssemblyLoadContext relies is only available for netcore and not net framework.  Ideally we would want to do something like the msbuild example, using #if preprocessor directives and choosing to use Assembly.LoadFrom or AssemblyLoadContext depending on the framework.  However since Microsoft.Spark library is targeting netstandard2.0 , i'm not sure how to go about doing this.   Would you have any suggestions on how we can do this ?

Reply:
I think there are 2 possibilities:

1. Detect at runtime if we are running on .NET Framework - which you can use https://source.dot.net/#CoreFx.Private.TestUtilities/System/PlatformDetection.cs,24.
2. Multi-target Microsoft.Spark to build for both `net45` and `netcoreapp2.1`. But we would still need to target `netstandard2.0` as well so other `netstandard` libraries could reference Spark. This would involve more work and build time, since we would be compiling the assembly 3 times.

My suggestion would be to try (1) above, and fallback to (2) if necessary.

Reply:
@eerhardt thanks for your suggestions.

I've tried (1) and while it compiles, it fails at runtime, where it tries to load the incompatible library.  I suspect that the net461 Microsoft.Spark.Worker attempts to load it because Microsoft.Spark.dll still contains references to the library.

(2) worked, but we run into the concerns you brought up regarding building separate libraries targeting multiple frameworks.  What do you think about the following approach?

Add an AssemblyLoader.cs class to Microsoft.Spark (netstandard2.0) that looks like the following:
```csharp
    internal static class AssemblyLoader
    {
        internal static Func<string, Assembly> Loader { get; set; } = Assembly.Load;

        internal static Assembly LoadAssembly(string manifestModuleName)
        {
            var sep = Path.DirectorySeparatorChar;
            try
            {
                return Loader(
                    $"{Directory.GetCurrentDirectory()}{sep}{manifestModuleName}");
            }
            catch (FileNotFoundException)
            {
                return Loader(
                    $"{AppDomain.CurrentDomain.BaseDirectory}{sep}{manifestModuleName}");
            }
        }
```

In Microsoft.Spark.Worker, which targets (net461 and netcoreapp2.1), we can add `#if preprocessor directives` and set the above AssemblyLoader.Loader property to use AssemblyLoadContext if the target framework is netcore.

Reply:
I am loading the assembly by way of the delegate mentioned previously.  This seemed like the cleanest approach.